src/models/selection.js

```diff
@@ -621,23 +704,6 @@ EDGE_METHODS.forEach((pattern) => {
 })
 
 /**
- * Memoize read methods.
- */
-
-memoize(Selection.prototype, [
-  'hasAnchorAtStartOf',
```

Comment:
All these methods rely directly on methods from `Node` which are already memoized. This removes the overhead of memoization for `Selection`. It's rare to reuse the same `Selection` object, selections change often, whereas nodes don't.
In practice, I removed them when I was optimizing the rendering. `shouldComponentUpdate` does a lot of `hasEdgeIn`. The memoization was slowing everything down with no benefit. Without memoization, the rendering was faster.